display/handlers.go

```diff
@@ -78,9 +79,18 @@ func handleOther(u string) {
 	default:
 		// The config has a custom command to execute for URLs
 		fields := strings.Fields(handler)
-		err := exec.Command(fields[0], append(fields[1:], u)...).Start()
-		if err != nil {
-			Error("URL Error", "Error executing custom command: "+err.Error())
+		// safety check for semicolon to prevent url from escaping out of custom command
```

Comment:
There is no need for a safety check in my opinion. In most cases it will not be necessary, because the URL and other arguments are not being sent to the shell, they are being sent to a binary to use. I understand how in [your use case](https://github.com/makeworld-the-better-one/amfora/issues/212#issuecomment-804360604) a semicolon could mess things up, but I think that's a rare case, and additionally there are going to be many ways to try and escape a command, and it would be tedious, complex. and error-prone to try and enumerate all of them here. We don't even know what shell the user is using.

For your specific case, you should quote the URL: `gnome-terminal -- "myprogram --uri '{{url}}'"`

Reply:
I'm probably going to url encode specific special characters. i.e. ' " ; 
There has to be a safety check here.

Someone could easily wget an executable, then run the executable

Reply:
Hmm, this is getting tricky. But all you should need is to escape `'`. And then in the wiki and in `default-config.toml` we'll recommend using single quotes. The only way to do something malicious from within single quotes in bash is to close the quote and do something else, so you only need ot escape `'`. Right?

I feel like a simpler way of doing this needs to be found... One that isn't specific to shells.

Reply:
> I feel like a simpler way of doing this needs to be found... One that isn't specific to shells.

Found: https://linux.die.net/man/1/xdg-open
I spent some time trying to pwn myself this way. I haven't found a way.